bundle/regal/rules/bugs/inconsistent_args.rego

```diff
@@ -0,0 +1,63 @@
+# METADATA
+# description: Inconsistently named function arguments
+package regal.rules.bugs["inconsistent-args"]
+
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+import data.regal.ast
+import data.regal.result
+
+report contains violation if {
+	count(ast.functions) > 0
+
+	# Comprehension indexing — as made obvious here it would be great
+	# to have block level scoped ignore directives...
+	function_args_by_name := {name: args_list |
+		some i
+
+		# regal ignore:prefer-some-in-iteration
+		name := ast.ref_to_string(ast.functions[i].head.ref)
+		args_list := [args |
+			some j
+
+			# regal ignore:prefer-some-in-iteration
+			ast.ref_to_string(ast.functions[j].head.ref) == name
+
+			# regal ignore:prefer-some-in-iteration
+			args := ast.functions[j].head.args
+		]
+		count(args_list) > 1
+	}
+
+	some name, args_list in function_args_by_name
+
+	# "Partition" the args by their position
+	by_position := [s |
+		some i, _ in args_list[0]
+		s := [x | x := args_list[_][i]]
+	]
+
+	some position in by_position
+
+	inconsistent_args(position)
+
+	violation := result.fail(rego.metadata.chain(), result.location(find_function_by_name(name)))
+}
+
+inconsistent_args(position) if {
+	named_vars := {arg.value |
+		some arg in position
+		arg.type == "var"
+		not startswith(arg.value, "$")
+	}
+	count(named_vars) > 1
+}
+
+# Return the _second_ function found by name, as that
+# is reasonably the location the inconsistency is found
+find_function_by_name(name) := [fn |
```

Comment:
Unless we can a pinpoint the first function with an actual argument name inconsistency, isn't it more clear to just point to the first function declaration.

Reply:
Possibly, but the second function is at least arguably more likely to be inconsistent than the first :P